transformations/chinese_antonym_synonym_substitution/transformation.py

```diff
@@ -0,0 +1,107 @@
+import itertools
+import random
+import os
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+import jieba
+from nlpcda import Similarword
+
+def chinese_antonym_synonym_substitution(text,
+                          prob,
+                          seed,
+                          max_outputs,
+                          config,
+                          chinese_antonym_data
+                          ):
+    random.seed(seed)
+
+    perturbed_texts = []
+    word1_list = [chinese_antonym['word1'] for chinese_antonym in chinese_antonym_data]
+    word2_list = [chinese_antonym['word2'] for chinese_antonym in chinese_antonym_data]
+
+    if(config == 'synonym'):
+        smw = Similarword(create_num=max_outputs+1, change_rate=prob)
+        output_list = smw.replace(text)
+
+        # Notice that the first input is the non-perturb version
+        output_list = output_list[1:]
+        perturbed_texts = output_list
+
+    if(config == 'antonym'):
+
+        output = jieba.lcut(text)
+        for _ in itertools.repeat(None, max_outputs):
+            butter_text = ""
+            for perturb_word in output:
+                perturb_antonym = []
+                if(perturb_word in word1_list):
+                    indices = [index for index, word1 in enumerate(word1_list) if word1 == perturb_word]
+                    perturb_antonym = [word2_list[index] for index in indices]
+                elif(perturb_word in word2_list):
+                    indices = [index for index, word2 in enumerate(word2_list) if word2 == perturb_word]
+                    perturb_antonym = [word1_list[index] for index in indices]
+                else:
+                    print("Word to be perturbed is not in the Chinese Antonym database")
+
+                if random.random() <= prob and len(perturb_antonym) != 0:
+                    new_chinese_character = random.choice(perturb_antonym)
+                else:
+                    new_chinese_character = perturb_word
+                butter_text += new_chinese_character
+            perturbed_texts.append(butter_text)
+    return perturbed_texts
+
+def load_chinese_antonym_data():
+    dirname = os.path.dirname(__file__)
+    words_list = []
```

Comment:
Why not use a dictionary to store the antonyms? The keys are all the words, and the values are lists of antonyms for the key.
```
antonym_dict = {
    word1: [word1_antonym1, word1_antonym2, ...],
    word2: [...],
    word3: [...],
    ...
}
```
I think using a dict is more straightforward than having a list of word pairs, and in this way, you don't need to loop over the list to get the indices.

Reply:
Done!